custom_components/switch_humidifier/humidifier.py

```diff
@@ -205,7 +205,8 @@ def turn_off(self, **kwargs):
 
   def _state_changed(self, entity_id, old_state, new_state):
     """Called on sensor or switch state change"""
-    if not new_state is None and not new_state.state == 'unknown' and not new_state.state == 'unavailable':
+    if not new_state is None and not new_state.state == 'unknown' and not new_state.state == 'unavailable' and old_state.state != new_state.state:
```

Comment:
Can you use the same way of checking as in the rest, "not old_state.state == new_state.state" or replace the others "new_state.state != 'unavailable'".
Just  for unifying it.